### DISCUSSION TITLE
--parallel and git process

### DISCUSSION THREAD

**Discussion by jucian0** [vercel/turborepo#4259] [Help] [answered]
I have two workspaces, and I'm trying to run a git commit in every workspace by `git commit` command, but I'm receiving a git error:|
`Another git process seems to be running in this repository, e.g.`

It looks like the processes are running in parallel, am I right?

Question: Parallel should be false by default, so, this error should not happen, right?

**Comment by mehulkar** [comment, marked as answer]
> Question: Parallel should be false by default

You're correct that the [`--parallel` flag](https://turbo.build/repo/docs/reference/command-line-reference/run#--parallel) is false by default, but this setting does not mean that tasks do not run in parallel. `--parallel=true` means that the dependency graph will be ignored, so `--parallel=false` means that the dependency graph will _not_ be ignored. It is very confusing now that I think about it. 

In any case, the behavior you're seeing is expected. Turbo will attempt to run as many of your tasks in parallel as possible. The `--parallel` flag does not change this.

If you truly want to prevent _any_ tasks from running at the same time, you can use [`--concurrency=1`](https://turbo.build/repo/docs/reference/command-line-reference/run#--concurrency)
